source/vibe/core/file.d

```diff
@@ -736,7 +736,7 @@ struct FileStream {
 		//       be relied upon. For this reason, we MUST use the uninterruptible
 		//       version of asyncAwait here!
 		auto res = asyncAwaitUninterruptible!(FileIOCallback,
-			cb => eventDriver.files.read(m_fd, ctx.ptr, dst, mode, cb)
+			cb => eventDriver.files.read(m_fd, ctx.ptr, () @trusted { return dst; } (), mode, cb)
```

Comment:
Why the trusted ? Isn't this just essentially bypassing every scope checks ?

Reply:
There is no way around this, unfortunately. The `read` function invokes the callback asynchronously, so it definitely isn't `scope` - instead, `asyncAwaitUninterruptible` ensures that the callback gets invoked before it returns through its program logic. This looks like prime example where `@trusted` is needed.